tools/downloader/topologies.py

```diff
@@ -0,0 +1,104 @@
+import os
+import argparse
+
+from pathlib import Path
+
+import common
+import downloader
+import converter
+
+class Model:     
+    '''
+    {}
+    Model task type: {}
+
+    License: {}
+    '''
+
+    def __init__(self, model_name, *, precisions=None, download_dir=None):
+        '''
+        Downloads target model. If precisions is None, loads all available models.
+        If the model has already been downloaded, retrieves the model from the cache.
+        Creates an object which stores information about the downloaded model.
+
+        Parameters
+        ----------
+            model_name
+                Target model name.
+            precisions
+                List of target model precisions. By default is None.
+            download_dir
+                Model download directory. By default creates a folder 'models' 
+                in current directory and downloads to it. 
+        '''
+        if not download_dir: 
+            download_dir = Path('.').resolve() / 'models'
+        cache_dir = Path(download_dir) / '.cache'
+
+        parser = argparse.ArgumentParser()
+        args = argparse.Namespace(all=False, list=None, name=model_name, print_all=False)   
+        topology = common.load_models_from_args(parser, args)[0]
+
+        model_dir = download_dir / topology.subdirectory
+
+        flags = ['--name=' + model_name,
+                 '--output_dir=' + str(download_dir).replace('\\', '\\\\'), 
+                 '--cache_dir=' + str(cache_dir).replace('\\', '\\\\')]
+        if precisions:
+            flags += ['--precisions=' + ','.join([str(elem) for elem in precisions]) ]
```

Comment:
Now it takes a list of precision values. We should use it like:
```python
ssd = omz.Model('face-detection-0100', precisions=['FP16','FP32'])
```

Reply:
It seems that this is no longer the case. Why did you change it to a string? Structured data should be passed via a structured type.

Reply:
I changed it to a string because for ```downloader``` we should pass ```--precision```  as comma separated string of values.

Please specify how do you want the user to pass input precision values:
1. ```ssd = omz.Model('mobilenet-ssd', precisions='FP16,FP32')```
or
2. ```ssd = omz.Model('mobilenet-ssd', precisions=['FP16', 'FP32'])```

Reply:
I think 2 makes the most sense.